core/chains/evm/logpoller/log_poller.go

```diff
@@ -282,14 +289,17 @@ func (lp *logPoller) RegisterFilter(ctx context.Context, filter Filter) error {
 			lp.lggr.Warnw("Filter already present, no-op", "name", filter.Name, "filter", filter)
 			return nil
 		}
-		lp.lggr.Warnw("Updating existing filter with more events or addresses", "name", filter.Name, "filter", filter)
+		lp.lggr.Warnw("Updating existing filter", "name", filter.Name, "filter", filter)
 	}
 
 	if err := lp.orm.InsertFilter(ctx, filter); err != nil {
 		return pkgerrors.Wrap(err, "error inserting filter")
 	}
 	lp.filters[filter.Name] = filter
 	lp.filterDirty = true
+	if filter.MaxLogsKept > 0 {
+		lp.countBasedLogPruningActive.Store(true)
```

Comment:
Is not it possible that a filter with `MaxLogsKept > 0` will be loaded from DB and we won't set countBasedLogPruningActive to true?

Reply:
Yikes! Yes, a big oversight on my part. @mateusz-sekara just left a comment expressing skepticism about having this flag at all... so I guess I need to decide whether to just back out this change or add a check where `LoadFilters()` is called so that it always gets set if needed.

I'm kind of on the fence, I guess what I'll do is run it with `MaxLogsKept=0` today and see how fast the query is when `MaxLogsKept=0` for all filters and decide based on that.